src/arch/x86/x86_64/src/bootblock.S

```diff
@@ -196,37 +204,146 @@ _estart16bit:
  *	expose the _start symbol for bootblock.
  */
 	.align	4
-.globl _start32
-_start32:
-
-	lgdt	%cs:gdtptr
-	ljmp	$8, $__protected_start
-
 __protected_start:
+	// Enter here in 32-bit protected mode. Welcome to 1982.
 	/* Save the BIST value */
 	movl	%eax, %ebp
 
 	//post_code(POST_ENTER_PROTECTED_MODE)
 
-
 	movw	$16, %ax
 	movw	%ax, %ds
 	movw	%ax, %es
 	movw	%ax, %ss
 	movw	%ax, %fs
 	movw	%ax, %gs
-
+	// The top two stack pages are %cr3
+	movl $0x7e000, %esp
+	movl %esp, %cr3
 	/* Restore the BIST value to %eax */
 	movl	%ebp, %eax
 	
 	/* BIST result in eax */
 	mov	%eax, %ebx
 	/* entry64.inc preserves ebx. */
-#include <cpu/x86/64bit/entry64.inc>
+	//WTF#include <cpu/x86/64bit/entry64.inc>
 	mov	%ebx, %eax
+//1:	 jmp 1b
+	// Now for the big fun: Long Mode.
+	// Once again we put the data structures inline in this
+	// memory. This code is, we hope, PIC.
+	call 2f
```

Comment:
Please explain this trick. I understand it but leave itin comments for posterity so people don't get confused.